### ISSUE TITLE
Dragging colour select to sidebar makes it crash

### ISSUE THREAD

**Issue by writeblankspace** [xournalpp/xournalpp#4505]
(Please complete the following information, and then delete this line)

**Affects versions :**
 - OS: Linux Ubuntu 22.04 LTS
 - (Linux only) Desktop environment: Gnome-Wayland
 - Which version of libgtk do you use: 3.24.33
 - Version of Xournal++: 1.1.3+dev (nogitfound)
 - Installation method: used the command line for nightly (`sudo add-apt-repository ppa:andreasbutti/xournalpp-master`)

**Describe the bug**
Dragging the colour select to the left sidebar makes the app crash.

**To Reproduce**
1. Open Xournal++
2. Go to View > Toolbars > select Portrait > select customise
3. Drag color select tool to the left sidebar
4. Xournal++ crashes 

EDIT: dragging color select tool to any toolbar causes a crash. However, dragging it to the 'customise toolbars' window doesn't cause a crash.

**Expected behavior**
Color select tool goes to the left sidebar

**Additional context**
When running from terminal, these come up:
```
** Message: 11:21:54.514: Removed tool item COLOR_SELECT from Toolbar toolbarTop2 ID 528
** Message: 11:21:54.515: ToolbarData::insertItem(toolbarLeft2, COLOR_SELECT, 0);

** (xournalpp:8750): WARNING **: 11:21:54.527: [Crash Handler] Crashed with signal 11

** (xournalpp:8750): WARNING **: 11:21:54.527: [Crash Handler] Wrote crash log to: /home/username/.cache/xournalpp/errorlogs/errorlog.20221214-112154.log

** (xournalpp:8750): WARNING **: 11:21:54.644: Trying to emergency save the current open document…

** (xournalpp:8750): WARNING **: 11:21:54.644: Successfully saved document to "/home/username/.config/xournalpp/emergencysave.xopp"
```

Content of error logs:
```
Date: Wed Dec 14 11:21:54 2022
Error: signal 11
[bt]: (0) xournalpp(+0xb55f8) [0x55ff09fc95f8]
[bt]: (1) /lib/x86_64-linux-gnu/libc.so.6(+0x42520) [0x7f2561662520]
[bt]: (2) /lib/x86_64-linux-gnu/libgtk-3.so.0(gtk_widget_get_allocated_width+0x15) [0x7f25626e1f55]
[bt]: (3) xournalpp(_ZN16ColorSelectImage10drawWidgetEP6_cairo+0x4d) [0x55ff0a0b003d]
[bt]: (4) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x3e6eb8) [0x7f256271deb8]
[bt]: (5) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x3900e1) [0x7f25626c70e1]
[bt]: (6) /lib/x86_64-linux-gnu/libgobject-2.0.so.0(g_signal_emit_valist+0x1290) [0x7f2561f7f700]
[bt]: (7) /lib/x86_64-linux-gnu/libgobject-2.0.so.0(g_signal_emit+0x93) [0x7f2561f7f863]
[bt]: (8) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x39fd82) [0x7f25626d6d82]
[bt]: (9) /lib/x86_64-linux-gnu/libgtk-3.so.0(gtk_container_propagate_draw+0x235) [0x7f25624b7825]
[bt]: (10) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x18092d) [0x7f25624b792d]
[bt]: (11) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x13c391) [0x7f2562473391]
[bt]: (12) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x1782e1) [0x7f25624af2e1]
[bt]: (13) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x18c852) [0x7f25624c3852]
[bt]: (14) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x139705) [0x7f2562470705]
[bt]: (15) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x39fb64) [0x7f25626d6b64]
[bt]: (16) /lib/x86_64-linux-gnu/libgtk-3.so.0(gtk_container_propagate_draw+0x235) [0x7f25624b7825]
[bt]: (17) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x18092d) [0x7f25624b792d]
[bt]: (18) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x39fb64) [0x7f25626d6b64]
[bt]: (19) /lib/x86_64-linux-gnu/libgtk-3.so.0(gtk_container_propagate_draw+0x235) [0x7f25624b7825]
[bt]: (20) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x34bb9d) [0x7f2562682b9d]
[bt]: (21) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x1782e1) [0x7f25624af2e1]
[bt]: (22) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x18c852) [0x7f25624c3852]
[bt]: (23) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x3479e5) [0x7f256267e9e5]
[bt]: (24) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x39fb64) [0x7f25626d6b64]
[bt]: (25) /lib/x86_64-linux-gnu/libgtk-3.so.0(gtk_container_propagate_draw+0x235) [0x7f25624b7825]
[bt]: (26) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x18092d) [0x7f25624b792d]
[bt]: (27) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x122571) [0x7f2562459571]
[bt]: (28) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x1782e1) [0x7f25624af2e1]
[bt]: (29) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x18c852) [0x7f25624c3852]
[bt]: (30) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x122e15) [0x7f2562459e15]
[bt]: (31) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x39fb64) [0x7f25626d6b64]
[bt]: (32) /lib/x86_64-linux-gnu/libgtk-3.so.0(gtk_container_propagate_draw+0x235) [0x7f25624b7825]
[bt]: (33) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x18092d) [0x7f25624b792d]
[bt]: (34) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x122571) [0x7f2562459571]
[bt]: (35) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x1782e1) [0x7f25624af2e1]
[bt]: (36) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x18c852) [0x7f25624c3852]
[bt]: (37) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x122e15) [0x7f2562459e15]
[bt]: (38) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x39fb64) [0x7f25626d6b64]
[bt]: (39) /lib/x86_64-linux-gnu/libgtk-3.so.0(gtk_container_propagate_draw+0x235) [0x7f25624b7825]
[bt]: (40) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x18092d) [0x7f25624b792d]
[bt]: (41) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x39fb64) [0x7f25626d6b64]
[bt]: (42) /lib/x86_64-linux-gnu/libgtk-3.so.0(gtk_container_propagate_draw+0x235) [0x7f25624b7825]
[bt]: (43) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x18092d) [0x7f25624b792d]
[bt]: (44) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x3c060e) [0x7f25626f760e]
[bt]: (45) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x39fb64) [0x7f25626d6b64]
[bt]: (46) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x3abdb1) [0x7f25626e2db1]
[bt]: (47) /lib/x86_64-linux-gnu/libgtk-3.so.0(gtk_main_do_event+0xd5b) [0x7f256258954b]
[bt]: (48) /lib/x86_64-linux-gnu/libgdk-3.so.0(+0x39743) [0x7f2562269743]
[bt]: (49) /lib/x86_64-linux-gnu/libgdk-3.so.0(+0x4c151) [0x7f256227c151]
[bt]: (50) /lib/x86_64-linux-gnu/libgdk-3.so.0(+0x51221) [0x7f2562281221]
[bt]: (51) /lib/x86_64-linux-gnu/libgdk-3.so.0(+0x51418) [0x7f2562281418]
[bt]: (52) /lib/x86_64-linux-gnu/libgobject-2.0.so.0(g_signal_emit_valist+0x1290) [0x7f2561f7f700]
[bt]: (53) /lib/x86_64-linux-gnu/libgobject-2.0.so.0(g_signal_emit+0x93) [0x7f2561f7f863]
[bt]: (54) /lib/x86_64-linux-gnu/libgdk-3.so.0(+0x46c6f) [0x7f2562276c6f]
[bt]: (55) /lib/x86_64-linux-gnu/libgdk-3.so.0(+0x332ad) [0x7f25622632ad]
[bt]: (56) /lib/x86_64-linux-gnu/libglib-2.0.so.0(+0x562c8) [0x7f2561e692c8]
[bt]: (57) /lib/x86_64-linux-gnu/libglib-2.0.so.0(g_main_context_dispatch+0x174) [0x7f2561e68c44]
[bt]: (58) /lib/x86_64-linux-gnu/libglib-2.0.so.0(+0xaa6c8) [0x7f2561ebd6c8]
[bt]: (59) /lib/x86_64-linux-gnu/libglib-2.0.so.0(g_main_context_iteration+0x33) [0x7f2561e663e3]
[bt]: (60) /lib/x86_64-linux-gnu/libgio-2.0.so.0(g_application_run+0x1a5) [0x7f25620bdfb5]
[bt]: (61) xournalpp(_ZN11XournalMain3runEiPPc+0x5df) [0x55ff09fc75ef]
[bt]: (62) /lib/x86_64-linux-gnu/libc.so.6(+0x29d90) [0x7f2561649d90]
[bt]: (63) /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0x80) [0x7f2561649e40]
[bt]: (64) xournalpp(_start+0x25) [0x55ff09fc5815]


Try to get a better stracktrace...
[bt] #1 xournalpp(+0xb5a9d) [0x55ff09fc9a9d]
??:0
[bt] #2 /lib/x86_64-linux-gnu/libc.so.6(+0x42520) [0x7f2561662520]
??:0
[bt] #3 /lib/x86_64-linux-gnu/libgtk-3.so.0(gtk_widget_get_allocated_width+0x15) [0x7f25626e1f55]
??:0
[bt] #4 xournalpp(_ZN16ColorSelectImage10drawWidgetEP6_cairo+0x4d) [0x55ff0a0b003d]
??:0
[bt] #5 /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x3e6eb8) [0x7f256271deb8]
??:0
[bt] #6 /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x3900e1) [0x7f25626c70e1]
??:0
[bt] #7 /lib/x86_64-linux-gnu/libgobject-2.0.so.0(g_signal_emit_valist+0x1290) [0x7f2561f7f700]
??:0
[bt] #8 /lib/x86_64-linux-gnu/libgobject-2.0.so.0(g_signal_emit+0x93) [0x7f2561f7f863]
??:0
[bt] #9 /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x39fd82) [0x7f25626d6d82]
??:0
[bt] #10 /lib/x86_64-linux-gnu/libgtk-3.so.0(gtk_container_propagate_draw+0x235) [0x7f25624b7825]
??:0
[bt] #11 /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x18092d) [0x7f25624b792d]
??:0
[bt] #12 /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x13c391) [0x7f2562473391]
??:0
[bt] #13 /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x1782e1) [0x7f25624af2e1]
??:0
[bt] #14 /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x18c852) [0x7f25624c3852]
??:0
[bt] #15 /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x139705) [0x7f2562470705]
??:0
[bt] #16 /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x39fb64) [0x7f25626d6b64]
??:0
[bt] #17 /lib/x86_64-linux-gnu/libgtk-3.so.0(gtk_container_propagate_draw+0x235) [0x7f25624b7825]
??:0
[bt] #18 /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x18092d) [0x7f25624b792d]
??:0
[bt] #19 /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x39fb64) [0x7f25626d6b64]
??:0
[bt] #20 /lib/x86_64-linux-gnu/libgtk-3.so.0(gtk_container_propagate_draw+0x235) [0x7f25624b7825]
??:0
[bt] #21 /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x34bb9d) [0x7f2562682b9d]
??:0
[bt] #22 /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x1782e1) [0x7f25624af2e1]
??:0
[bt] #23 /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x18c852) [0x7f25624c3852]
??:0
[bt] #24 /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x3479e5) [0x7f256267e9e5]
??:0
[bt] #25 /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x39fb64) [0x7f25626d6b64]
??:0
[bt] #26 /lib/x86_64-linux-gnu/libgtk-3.so.0(gtk_container_propagate_draw+0x235) [0x7f25624b7825]
??:0
[bt] #27 /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x18092d) [0x7f25624b792d]
??:0
[bt] #28 /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x122571) [0x7f2562459571]
??:0
[bt] #29 /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x1782e1) [0x7f25624af2e1]
??:0
[bt] #30 /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x18c852) [0x7f25624c3852]
??:0
[bt] #31 /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x122e15) [0x7f2562459e15]
??:0
```

**Comment by Technius** [comment]
The stack trace doesn't seem to have any useful information (neither does the debugger help). This appears to be some sort of memory management bug.

**Comment by jhilmer** [comment]
I have a fix for this one. I will create a PR shortly

**Comment by jhilmer** [comment]
Should be closed. Fix is in master.

**Comment by Technius** [comment]
Reminder to myself to check whether this bug was due to a regression or if it was already an existing bug.

**Comment by rolandlo** [comment]
> Reminder to myself to check whether this bug was due to a regression or if it was already an existing bug.

I checked it. In version 1.1.3 (I tested both the snap and flatpak) the app doesn't crash when dragging the color select icon to another sidebar, but there was already an issue that the color didn't update for the that icon after it had been dragged during the toolbar customization. So #4535 is partly a fix of a regression (the crash) and partly a fix of an existing issue.
